kcl/kcl.go

```diff
@@ -288,7 +317,7 @@ func (kclp *KCLProcess) Run() {
 		action, err := kclp.handleLine(line)
 		if err != nil {
 			kclp.ioHandler.writeError(fmt.Sprintf("ERR Handle line: %+#v", err))
-		} else if action == "shutdown" {
+		} else if action != nil && action.Name() == ActionNameShutdown {
```

Comment:
Are there cases when `action` will be nil?  If I'm reading the code correctly, that should only happen on error.

Reply:
Actually, it just occurred to me.  You may also have to add a "ActionNameShutdownRequested" check here as well.